picard/album.py

```diff
@@ -467,54 +472,48 @@ def _remove_file(self, track, file):
         file.metadata_images_changed.disconnect(self.update_metadata_images)
         remove_metadata_images(self, [file])
 
-    def match_files(self, files, use_recordingid=True):
+    def _match_files(self, files, recordingid=None, threshold=0):
         """Match files to tracks on this album, based on metadata similarity or recordingid."""
+        moves = defaultdict(list)
         for file in list(files):
             if file.state == File.REMOVED:
                 continue
             matches = []
-            recordingid = file.metadata['musicbrainz_recordingid']
-            if use_recordingid and mbid_validate(recordingid):
-                matches = self._get_recordingid_matches(file, recordingid)
+            recid = recordingid or file.metadata['musicbrainz_recordingid']
+            if mbid_validate(recid):
+                tracknumber = file.metadata['tracknumber']
+                discnumber = file.metadata['discnumber']
+                for track in self.tracks:
+                    tm = track.orig_metadata
+                    if recid == tm['musicbrainz_recordingid']:
+                        if tracknumber == tm['tracknumber']:
+                            if discnumber == tm['discnumber']:
+                                heappush(matches, (-4.0, track))
+                                break
+                            else:
+                                heappush(matches, (-3.0, track))
+                        else:
+                            heappush(matches, (-2.0, track))
```

Comment:
Any specific reason why you did not leave this code block in a separate method? I think that helped reducing the complexity of this method.

Reply:
> Any specific reason why you did not leave this code block in a separate method? I think that helped reducing the complexity of this method.

Because it is only used here, and skip a function call in a loop, i also see possible improvements in this stuff.